FlowCrypt/Functionality/Services/Compose Message Service/ComposeMessageService.swift

```diff
@@ -217,105 +217,105 @@ final class ComposeMessageService {
 
         for attachment in message.atts {
             guard let data = Data(base64Encoded: attachment.base64) else { continue }
 
-            let encryptedFile = try await core.encryptFile(
-                pubKeys: message.pubKeys,
-                fileData: data,
-                name: attachment.name
+            let encryptedFile = try await core.encrypt(
+                file: data,
+                name: attachment.name,
+                pubKeys: message.pubKeys
             )
             let encryptedAttachment = SendableMsg.Attachment(
                 name: "\(attachment.name).pgp",
                 type: "application/pgp-encrypted",
-                base64: encryptedFile.encryptedFile.base64EncodedString()
+                base64: encryptedFile.base64EncodedString()
             )
             encryptedAttachments.append(encryptedAttachment)
         }
 
-        let passwordMessage = SendableMsg(
-            text: html,
-            html: html,
-            to: message.to,
-            cc: message.cc,
-            bcc: message.bcc,
-            from: message.from,
-            subject: message.subject,
-            replyToMimeMsg: message.replyToMimeMsg,
-            atts: [bodyAttachment] + encryptedAttachments,
-            pubKeys: nil,
-            signingPrv: nil,
-            password: nil
-        )
-        return try await core.composeEmail(msg: passwordMessage, fmt: .plain)
+        return encryptedAttachments
     }
 
-    func prepareAndUploadPwdEncryptedMsg(message: SendableMsg, replyToken: String) async throws -> String {
-        let formatted = try await formatPasswordMessage(
+    private func prepareAndUploadPwdEncryptedMsg(message: SendableMsg) async throws -> String {
+        let replyToken = try await enterpriseServer.getReplyToken(for: message.from)
+
+        let bodyWithReplyToken = try getPwdMsgBodyWithReplyToken(
             message: message,
             replyToken: replyToken
         )
-
-        let formattedMessage = SendableMsg(
-            text: formatted.mimeEncoded.toStr(),
-            html: formatted.mimeEncoded.toStr(),
-            to: message.to,
-            cc: message.cc,
-            bcc: message.bcc,
-            from: message.from,
-            subject: message.subject,
-            replyToMimeMsg: message.replyToMimeMsg,
+        let msgWithReplyToken = message.copy(
+            body: bodyWithReplyToken,
             atts: message.atts,
-            pubKeys: [],
-            signingPrv: nil,
-            password: message.password
+            pubKeys: nil
+        )
+        let pgpMimeWithAttachments = try await core.composeEmail(
+            msg: msgWithReplyToken,
+            fmt: .plain
+        ).mimeEncoded.toStr()
+
+        let sendableMsg = message.copy(
+            body: SendableMsgBody(
+                text: pgpMimeWithAttachments,
+                html: pgpMimeWithAttachments
+            ),
+            atts: [],
+            pubKeys: []
         )
```

Comment:
This is an unneeded step. You should be able to pass `pgpMimeWithAttachments` (as `Data` without `.toStr()`) directly into `core.encrypt`. In fact, you could/should use `core.encryptFile` equivalent below.